nidirect_landing_pages/nidirect_landing_pages.module

```diff
@@ -0,0 +1,217 @@
+    // This will be a link to a taxonomy term.
+    $results[strtolower($row->_entity->getName())] = [
+      'type' => 'term_teaser',
+      'field' => 'field_term',
+      'target_id' => $row->tid,
+    ];
+  }
+  return $results;
+}
+
+/**
+ * Implements hook_page_attachments().
+ */
+function nidirect_landing_pages_page_attachments(array &$attachments) {
+  // Add landing page js if we are creating or editing
+  // a landing page node.
+  $add_js = FALSE;
+  if (\Drupal::service('router.admin_context')->isAdminRoute()) {
+    if (\Drupal::request()->server->get('REQUEST_URI') == '/node/add/landing_page') {
+      $add_js = TRUE;
+    }
+    else {
+      $node = \Drupal::routeMatch()->getParameter('node');
+      if (!empty($node) && ($node->getType() == 'landing_page')) {
+        $add_js = TRUE;
+      }
+    }
+  }
+  if ($add_js) {
+    $attachments['#attached']['library'][] = 'nidirect_landing_pages/landing_listing';
+  }
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function nidirect_landing_pages_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  // Add description to theme/subtheme field for landing pages.
+  if (($form_id == 'node_landing_page_form') || ($form_id == 'node_landing_page_edit_form')) {
+    if (isset($form['field_subtheme'])) {
+      $msg = "Note that the theme/subtheme that you choose here will be overridden by this landing page. <br/>";
+      $msg .= "As an example, if you select 'Motoring / Road Safety' here then when you visit the page at ";
+      $msg .= "/information-and-services/motoring there will be a link to this page labelled 'Road Safety'.";
+      $form['field_subtheme']['widget']['#description'] = t($msg);
+    }
+    // Only show 'Listing' field if 'manually control listing'
+    // checkbox has been checked.
+    if (isset($form['field_listing'])) {
+      //$form['field_listing']['#states'] = ['visible' => [':input[name="field_manually_control_listing[value]"]' => ['checked' => TRUE]]];
+    }
+    // Add ajax to the 'manually control listing' checkbox.
+    if (isset($form['field_manually_control_listing'])) {
+      $form['field_manually_control_listing']['widget']['value']['#ajax'] = [
+        'callback' => 'nidirect_landing_pages_populate_listing',
+        'event' => 'change',
+        'wrapper' => 'edit-field-listing-wrapper',
+      ];
+    }
+    // Add a validate handler to detect any duplicates.
+    $form['#validate'][] = 'nidirect_landing_pages_form_validate';
+  }
+}
+
+/**
+ * Utility Ajax function.
+ */
+function nidirect_landing_pages_populate_listing(array &$form, FormStateInterface $form_state) {
+  // The main purpose of this is to make sure that
+  // nidirect_common_node_prepare_form is called.
+  if (!$form_state->getValue('field_manually_control_listing')['value']) {
+    // If 'manual control' checkbox is unchecked, make sure
+    // that listing field is not shown.
+    $form['field_listing']['#access'] = FALSE;
```

Comment:
Do you want this to be 'hidden' instead? I've had a few instances where I've tried this approach and when handling the form submission I can't get at the value because '#access = FALSE' not only hides the form element but effectively removes it from the form's payload when submitted. If this works, that's great... feel free to disregard 👍